CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -178,16 +193,20 @@ class SiStripGainFromCalibTree : public ConditionDBWriter<SiStripApvGain> {
 	std::string  OutputGains;
 	vector<string> VInputFiles;
 
-	MonitorElement*        Charge_Vs_Index;
-	MonitorElement*        Charge_Vs_Index_Absolute;
-	MonitorElement*        Charge_Vs_PathlengthTIB;
-	MonitorElement*        Charge_Vs_PathlengthTOB;
-	MonitorElement*        Charge_Vs_PathlengthTIDP;
-	MonitorElement*        Charge_Vs_PathlengthTIDM;
-	MonitorElement*        Charge_Vs_PathlengthTECP1;
-	MonitorElement*        Charge_Vs_PathlengthTECP2;
-	MonitorElement*        Charge_Vs_PathlengthTECM1;
-	MonitorElement*        Charge_Vs_PathlengthTECM2;
+        enum statistic_type {None=-1, StdBunch, StdBunch0T, FaABunch, FaABunch0T, IsoBunch, IsoBunch0T, Harvest};
+
+        std::vector<string>    dqm_tag_;
```

Comment:
@davidlange6

I put a reserve statement for dam_tag_ in the constructor. 

Sorry I was confused among std::arrays and  C arrays. I think that using std::array is fine, I will change the code in the next update. Meanwhile I didn’t to delay too much this PR, so I put the reserve statement, because it was the fastest thing to do.


> On 23 Mar 2016, at 12:06, David Lange <notifications@github.com> wrote:
> 
> In CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc <https://github.com/cms-sw/cmssw/pull/13724#discussion_r57141693>:
> 
> > @@ -178,16 +193,20 @@ class SiStripGainFromCalibTree : public ConditionDBWriter<SiStripApvGain> {
> >  	std::string  OutputGains;
> >  	vector<string> VInputFiles;
> >  
> > -	MonitorElement*        Charge_Vs_Index;
> > -	MonitorElement*        Charge_Vs_Index_Absolute;
> > -	MonitorElement*        Charge_Vs_PathlengthTIB;
> > -	MonitorElement*        Charge_Vs_PathlengthTOB;
> > -	MonitorElement*        Charge_Vs_PathlengthTIDP;
> > -	MonitorElement*        Charge_Vs_PathlengthTIDM;
> > -	MonitorElement*        Charge_Vs_PathlengthTECP1;
> > -	MonitorElement*        Charge_Vs_PathlengthTECP2;
> > -	MonitorElement*        Charge_Vs_PathlengthTECM1;
> > -	MonitorElement*        Charge_Vs_PathlengthTECM2;
> > +        enum statistic_type {None=-1, StdBunch, StdBunch0T, FaABunch, FaABunch0T, IsoBunch, IsoBunch0T, Harvest};
> > +
> > +        std::vector<string>    dqm_tag_;
> for arrays - its memory overhead (obviously minor) and bounds checking for free. Reserve is certainly needed for your vectors otherwise.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/cms-sw/cmssw/pull/13724/files/588628c5918e3dc4da22ac8b6a0533d05b2fb855#r57141693>